runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -2941,6 +3007,82 @@ void set_hugepage_info(void)
 }
 
 
+static
+void install_SIGBUS_handler(void)
+{
+  struct sigaction act;
+
+  act.sa_flags = SA_RESTART | SA_SIGINFO;
+  if (sigemptyset(&act.sa_mask) != 0) {
```

Comment:
Hmm.  My copy of ``sigaction(2)`` says after describing ``sa_mask``: "In addition, the signal which triggered the handler will be blocked, unless teh ``SA_NODEFER`` flag is used."  So I think with this setup, ``SIGBUS`` will be blocked in the handler, and I don't think the ``raise()`` will have an effect until return from the (intended-to-be-outer) handler.